gitlab/utils.py

```diff
@@ -31,6 +32,31 @@ def __call__(self, chunk: Any) -> None:
         print(chunk)
 
 
+class MaskingFormatter(logging.Formatter):
+    """A logging formatter that can mask credentials"""
+
+    def __init__(
+        self,
+        fmt: Optional[str] = logging.BASIC_FORMAT,
+        datefmt: Optional[str] = None,
+        style: str = "%",  # should be Literal, but needs extensions for 3.7
+        validate: bool = True,
+        masked: Optional[str] = None,
+    ) -> None:
+        super().__init__(fmt, datefmt, style, validate)  # type: ignore[arg-type]
+        self.masked = masked
```

Comment:
I worked around the failure in Python 3.7 in #2205

https://github.com/python-gitlab/python-gitlab/pull/2205/files#diff-f8491e1894d267d0ea4bcd44eaf0e9313e2cd3d7a7a2a32bcbbb0dcba902b25eR38-R45